### DISCUSSION TITLE
Origins of mods on kexquake AWS that are not at Bethesda

### DISCUSSION THREAD

**Discussion by MartinHowe426** [andrei-drexler/ironwail#296] [Q&A] [answered]
There are several mods available in the mods menu that are not listed in the original remaster engine's menus (at least as of a couple weeks ago) but are listed in Ironwail's; for example, ttb ("Time to Belong"); the url says they are an AWS bucket with kexquake in the url.

What are these mods; I mean if they're not officially remastered content from Bethesda, then what are they? Just the originals? And in the latter case, are they not all the latest versions? TTB, for example, has a version 1.01 from May 2023 that fixes a few bugs, but it has a different size than the one Ironwail downloads, which the JSON file says is 1.0, dated the day before.

As well as knowing why these are there, before going on a hunt for the latest versions of everything, I'd appreciate some guidance; thanks.

**Comment by alexey-lysiuk** [comment]
Ironwail downloads JSON file from unofficial add-ons server, parses it, and fills Mods menu with entries from the given file.
At the moment, it contains 26 mods:

- Beyond Belief
- Contract Revoked
- Dark Triad
- Deathmatch Dimension
- Elder World Jam
- Empire of Disorder
- Epochs of Enmity
- Euclid's Nightmare
- Honey
- IKSPQ
- Insomnia
- Koohoo Retro Jam
- Map Jam X: Insomnia
- Operation: Urth Majik
- Punishment Due
- QDOOM
- Quake 64
- Realm of Tiddles
- Rubicon 2
- Sacrilege
- Slave Zero X
- Spiritworld
- Squire of Time
- Terra
- Time to Belong
- Underdark Overbright & Copper

[Here](https://github.com/andrei-drexler/ironwail/blob/5f54a9880dc05ed0e6866368273273b516c50ffa/Quake/host_cmd.c#L1250) is an entry point function of this feature.

**Reply by andrei-drexler** [reply, marked as answer]
Just a small note - despite the official-sounding URL, the server is not actually *official* - it's maintained by members of the Quake community (@Immorpher and @neogeographica), and it does indeed contain all the official mods + a few extras. For more info, 
there's a [quick-start page](https://kexquake.s3.amazonaws.com/index.html), a [thread](https://www.slipseer.com/index.php?threads/quake-enhanced-add-ons-server-kex-quake-quake-rerelease.124/) on Slipseer, and a dedicated [Discord server](https://discord.gg/zg2ENYQDty).

As for the various versions of different add-ons that you might encounter - unfortunately IW doesn't currently support updating mods that have already been downloaded, or for that matter even just deleting them. There's definitely more work to be done on that front.

**Comment by Immorpher** [comment]
Indeed as @andrei-drexler said, they come from an unofficial mod server that we run. You can find more details here: https://kexquake.s3.amazonaws.com/index.html

Interestingly the Remaster (last I checked) doesn't have a way of updating downloaded mods either. I dont think they get updated much if at all. But we will update the mods on our end occasionally if needed. Empire of Disorder had a few bug fixes. I believe Time to Belong had an update to its lightmaps, where they were re-compiled at higher quality (and to consider transparent water).

**Comment by MartinHowe426** [comment]
Thanks very much, folks; it's a lot clearer now. I don't use the official remaster except to check for new official addons so this explanation helps a lot.
